src/js/common/schemaform/FormPage.jsx

```diff
@@ -79,17 +79,34 @@ class FormPage extends React.Component {
 
   render() {
     const { route } = this.props;
+    const form = this.props.form;
     const {
       data,
       schema,
       uiSchema
-    } = this.props.form[route.pageConfig.pageKey];
+    } = form[route.pageConfig.pageKey];
+
+    // Flatten the data from every page to pass to SchemaForm to eventually be
+    //  used in uiSchemaValidate()
+    // I'd rather not have this here, but console.time() tells me it's executing
+    //  locally in 1.49e+12ms on the edu 5495, which is admittedly a relatively
+    //  small form, but it shouldn't be a significant hit at any rate.
+    const formData = Object.keys(form).reduce((carry, pageName) => {
```

Comment:
How about renaming `data` above to `pageData` so that there is no `data` variable.

Also, should we add a flatten helper and use it both here and in `validation.js`?